cnf-tests/testsuites/e2esuite/test_suite_test.go

```diff
@@ -4,75 +4,79 @@
 	_ "github.com/metallb/metallb-operator/test/e2e/functional/tests"
 	_ "github.com/openshift/ptp-operator/test/conformance/ptp"
 
-	perfUtils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
-
-	sriovClean "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/clean"
-	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/clean"
-	testclient "github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
-	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
-	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/features"
 	testutils "github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/utils"
-	perfClean "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/clean"
-	ptpClean "github.com/openshift/ptp-operator/test/utils/clean"
 
-	numaserialconf "github.com/openshift-kni/numaresources-operator/test/e2e/serial/config"
 	_ "github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests"
 
-	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	ginkgo_reporters "kubevirt.io/qe-tools/pkg/ginkgo-reporters"
 )
 
 // TODO: we should refactor tests to use client from controller-runtime package
 // see - https://github.com/openshift/cluster-api-actuator-pkg/blob/master/pkg/e2e/framework/framework.go
 
 var (
-	junitPath          *string
-	reportPath         *string
-	skipTestNSCreation bool
+	junitPath  *string
+	reportPath *string
 )
 
+var suiteFixtureMap = map[string]features.SuiteFixture{
+	"performance":   &features.PAOFixture{},
+	"sriov":         &features.SriovFixture{},
+	"dpdk":          &features.DPDKFixture{},
+	"gatekeeper":    &features.GatekeeperFixture{},
+	"sro":           &features.SROFixture{},
+	"numaresources": &features.NumaresourcesFixture{},
+	"xt_u32":        &features.XTU32Fixture{},
+	"ptp":           &features.PTPFixture{},
+	"bondcni":       &features.BondcniFixture{},
+	"tuningcni":     &features.TuningcniFixture{},
+	"fec":           &features.FECFixture{},
+	"vrf":           &features.VRFFixture{},
+	"ovs_qos":       &features.OVSQOSFixture{},
+	"sctp":          &features.SCTPFixture{},
+}
+
 func init() {
 	junitPath = flag.String("junit", "junit.xml", "the path for the junit format report")
 	reportPath = flag.String("report", "", "the path of the report file containing details for failed tests")
 
-	skipTestNSCreation = false
-	if os.Getenv("SKIP_TEST_NAMESPACES_CREATION") == "true" {
-		skipTestNSCreation = true
+	featuresVar := os.Getenv("FEATURES")
+	if featuresVar != "" {
+		for feature := range suiteFixtureMap {
+			if !strings.Contains(featuresVar, feature) {
+				delete(suiteFixtureMap, feature)
+			}
+		}
 	}
```

Comment:
nit: can you please print here the features we are going to run? this can help for debugs

Reply:
these are not the features we are going to run, these are only the features (from the features we are going to run) that need a configuration.